keys/signer_secp256r1.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package keys
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+
+	"github.com/coinbase/rosetta-sdk-go/asserter"
+	"github.com/coinbase/rosetta-sdk-go/types"
+)
+
+// SignerSecp256r1 is initialized from a keypair
+type SignerSecp256r1 struct {
+	KeyPair *KeyPair
+}
+
+// The Ecdsa signature is the couple (r, s), both r and s are 32 bytes
+const (
+	EcdsaRLen   = 32
```

Comment:
small nit: should this be shared with secp256k1? The signing algorithm ECDSA is constant between the two I believe, so these constants should be shared ideally. We can tackle in a future PR though, it's a small nit

Reply:
Yes, this is the same in secp256k1, so I named it as EcdsaRLen, not Secp256r1RLen. Maybe I should put this to https://github.com/coinbase/rosetta-sdk-go/blob/23200e54f95116600ff4617be5b78e1768ba689c/keys/signer_secp256k1.go#L33 ? But I don't want to modify secp256k1 code for this PR is for secp256r1, and this is not used in secp256k1 code.

Reply:
That's fine, if it's not used in secp256k1 code then we can ignore it or tackle it later. It certainly is a great validation to check sig / message length